game-app/game-core/src/main/java/games/strategy/triplea/ui/StatPanel.java

```diff
@@ -437,7 +233,7 @@ class PuStat extends ResourceStat {
   private static Resource getResourcePUs(final GameData data) {
     final Resource pus;
     try (GameData.Unlocker ignored = data.acquireReadLock()) {
-      pus = data.getResourceList().getResource(Constants.PUS);
+      pus = data.getResourceList().getResource(Constants.PUS).orElse(null);
```

Comment:
@BBlaza Same, either the method assumes the value must be available, then new method `getResourceOrThrow?` or a default value should be used, e.g., `orElse(0)`.